### PR TITLE
Ra/e2e sobject definitions

### PR DESCRIPTION
**Pull request by RitamAgrawal**
### What does this PR do?
Implement E2E tests for refreshing sobject definitions.

### What issues does this PR fix or reference?

@W-12370898@

See also https://salesforce.quip.com/qVEkABuGd7ao and https://salesforce.quip.com/q3tDAvAb2Udq

### CODE REVIEW

File: test/specs/sObjectsDefinitions.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+    const successNotificationWasFound = await utilities.attemptToFindNotification(workbench, 'SFDX: Push Source to Default Scratch Org successfully ran', 10);
+    expect(successNotificationWasFound).toBe(true);
+
+    const outputPanelText = await utilities.attemptToFindOutputPanelText('Salesforce CLI', 'Starting SFDX: Push Source to Default Scratch Org', 5);
+    expect(outputPanelText).not.toBeUndefined();
+    expect(outputPanelText).toContain('Pushed Source');
+  });
+
+  step('Refresh SObject Definitions for Custom SObjects', async () => {
```
Comment (jeffb-sfdc) [inline test/specs/sObjectsDefinitions.e2e.ts:75]: The "Refresh SObject Definitions for Custom SObjects" test is failing for me on my local machine.

```
[chrome 102.0.5005.196 mac os x #0-0] Error: expect(received).not.toEqual(expected) // deep equality
[chrome 102.0.5005.196 mac os x #0-0]
[chrome 102.0.5005.196 mac os x #0-0] Expected: not undefined
[chrome 102.0.5005.196 mac os x #0-0]
[chrome 102.0.5005.196 mac os x #0-0]     at Context.<anonymous> (/Users/jbeeghly/src/github.com/forcedotcom/salesforcedx-vscode-automation-tests-3/test/specs/sObjectsDefinitions.e2e.ts:118:34)
```

File: test/specs/sObjectsDefinitions.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { step } from 'mocha-steps';
+import { DefaultTreeItem, TextEditor, TreeItem } from 'wdio-vscode-service';
```
Comment (jeffb-sfdc) [inline test/specs/sObjectsDefinitions.e2e.ts:8]: TextEditor is needed an can be removed.

File: test/specs/sObjectsDefinitions.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    scratchOrg = new ScratchOrg('sObjectsDefinitions', false);
+    await scratchOrg.setUp();
+    const projectPath = scratchOrg.projectFolderPath;
+    const tempFolderPath = scratchOrg.tempFolderPath;
+    const source = path.join(tempFolderPath!, '..', 'test', 'testData', 'CustomSObjects');
+    const destination = path.join(projectPath!, 'force-app', 'main', 'default', 'objects');
+
+    fs.cp(source, destination, { recursive: true }, async (error) => {
+      if (error) {
+        await scratchOrg.tearDown();
```
Comment (jeffb-sfdc) [inline test/specs/sObjectsDefinitions.e2e.ts:29]: If there is an error at this point, we should log the error.

File: test/specs/sObjectsDefinitions.e2e.ts (hunk 1 above)
Comment (RitamAgrawal) [reply test/specs/sObjectsDefinitions.e2e.ts:75]: I ran it and it's all passing on my local.

Comment (jeffb-sfdc) [reply test/specs/sObjectsDefinitions.e2e.ts:75]: When I run, within /.sfdx/tools there is [sobjects, soqlMetadata, testresults, apex.db]

But when `await toolsTreeItem.getChildren();` is called, only [testresults, apex.db] are returned.

**Review by jeffb-sfdc — APPROVED** [review]

**Pull request merged**